Atril Document Viewer

### Why is it worth to add this package?

It is Simple yet Very Nice all in one Document viewer as it supports PostScript, DJVU, PDF and many other files.

### Home page URL

_No response_

### Source code URL

https://github.com/mate-desktop/atril

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

### Additional information

_No response_

This package have low priority IMO, because atril is evince fork, but we have evince.

I started creating package, but I failed on build. Probably need to patch bcopy function

```
/home/builder/.termux-build/atril/src/backend/dvi/mdvi-lib/dviread.c:1338:2: error: call to undeclared function 'bcopy'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        memcpy(&dvi->stack[dvi->stacktop], &dvi->pos, sizeof(DviState));
        ^
/data/data/com.termux/files/usr/include/kpathsea/c-memstr.h:53:31: note: expanded from macro 'memcpy'
#define memcpy(to, from, len) bcopy ((from), (to), (len))
                              ^
/home/builder/.termux-build/atril/src/backend/dvi/mdvi-lib/dviread.c:1354:2: error: call to undeclared function 'bcopy'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        memcpy(&dvi->pos, &dvi->stack[dvi->stacktop-1], sizeof(DviState));
        ^
/data/data/com.termux/files/usr/include/kpathsea/c-memstr.h:53:31: note: expanded from macro 'memcpy'
#define memcpy(to, from, len) bcopy ((from), (to), (len))
```



```
TERMUX_PKG_HOMEPAGE=https://mate-desktop.org/
TERMUX_PKG_DESCRIPTION="Atril is a simple multi-page document viewer. It can display and print PostScript (PS), Encapsulated PostScript (EPS), DJVU, DVI and Portable Document Format (PDF) files."
TERMUX_PKG_LICENSE="GPL-2.0, LGPL-2.0"
TERMUX_PKG_MAINTAINER="@termux"
TERMUX_PKG_VERSION="1.26.1"
TERMUX_PKG_SRCURL=https://github.com/mate-desktop/atril/releases/download/v${TERMUX_PKG_VERSION}/atril-${TERMUX_PKG_VERSION}.tar.xz
TERMUX_PKG_SHA256=a53a613acb9701a18ad671bf590254d1ce836ba0ae1be2c0bd8948fdf6b491a4
TERMUX_PKG_AUTO_UPDATE=true
TERMUX_PKG_DEPENDS="libsecret, gtk3, libsm, libice, caja, texlive-bin, poppler"
```

--- a/x11-packages/atril/build.sh
+++ b/x11-packages/atril/build.sh
@@ -1,0 +1,32 @@
+TERMUX_PKG_HOMEPAGE=https://mate-desktop.org
+TERMUX_PKG_DESCRIPTION="MATE document viewer"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="1.28.0"
+TERMUX_PKG_SRCURL=https://pub.mate-desktop.org/releases/${TERMUX_PKG_VERSION%.*}/atril-${TERMUX_PKG_VERSION}.tar.xz
+TERMUX_PKG_SHA256=ced4725f6e9b71c4ea63676bfc3cc3be09d29dba08aa7a7ab97964e0b4355162
+TERMUX_PKG_AUTO_UPDATE=true
+# links with poppler-glib, not poppler
+TERMUX_PKG_DEPENDS="atk, djvulibre, gdk-pixbuf, glib, gtk3, harfbuzz, libarchive, libc++, libcairo, libice, libsecret, libsm, libsoup3, libtiff, libxml2, mate-desktop, pango, poppler, texlive-bin, webkit2gtk-4.1, zlib"
+TERMUX_PKG_BUILD_DEPENDS="g-ir-scanner, glib-cross"
+TERMUX_PKG_DISABLE_GIR=false
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+--disable-caja
+--enable-djvu
+--enable-dvi
+--enable-epub
+--enable-introspection
+--enable-pixbuf
+"
+
+termux_step_pre_configure() {
+	TERMUX_PKG_VERSION=. termux_setup_gir
+
+	CPPFLAGS+=" -DHAVE_MEMCPY -Wno-deprecated-declarations"
+
+	# fix arm build and potentially other archs hidden bugs
+	# ERROR: ./lib/atril/3/backends/libpdfdocument.so contains undefined symbols:
+	# 162: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_idivmod
+	# 163: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_idiv
+	LDFLAGS+=" $($CC -print-libgcc-file-name)"
+}

--- a/x11-packages/atril/gir/AtrilDocument-1.5.0.xml
+++ b/x11-packages/atril/gir/AtrilDocument-1.5.0.xml
@@ -1,0 +1,288 @@
+<?xml version="1.0"?>
+<dump>
+  <boxed name="EvDocumentInfo" get-type="ev_document_info_get_type"/>
+  <boxed name="EvDocumentLicense" get-type="ev_document_license_get_type"/>
+  <class name="EvPage" get-type="ev_page_get_type" parents="GObject">
+  </class>
+  <class name="EvRenderContext" get-type="ev_render_context_get_type" parents="GObject">
+  </class>
+  <class name="EvDocument" get-type="ev_document_get_type" parents="GObject" abstract="1">
+  </class>
+  <boxed name="EvRectangle" get-type="ev_rectangle_get_type"/>
+  <boxed name="EvSourceLink" get-type="ev_source_link_get_type"/>
+  <class name="EvAttachment" get-type="ev_attachment_get_type" parents="GObject">
+    <property name="name" type="gchararray" flags="10" default-value="NULL"/>
+    <property name="description" type="gchararray" flags="10" default-value="NULL"/>
+    <property name="mtime" type="gulong" flags="10" default-value="0"/>
+    <property name="ctime" type="gulong" flags="10" default-value="0"/>
+    <property name="size" type="guint" flags="10" default-value="0"/>
+    <property name="data" type="gpointer" flags="10"/>
+  </class>
+  <class name="EvAnnotation" get-type="ev_annotation_get_type" parents="GObject" abstract="1">
+    <property name="page" type="EvPage" flags="10"/>
+    <property name="contents" type="gchararray" flags="3" default-value="NULL"/>
+    <property name="name" type="gchararray" flags="3" default-value="NULL"/>
+    <property name="modified" type="gchararray" flags="3" default-value="NULL"/>
+    <property name="color" type="gpointer" flags="3"/>
+    <property name="rgba" type="GdkRGBA" flags="227"/>
+  </class>
+  <interface name="EvAnnotationMarkup" get-type="ev_annotation_markup_get_type">
+    <prerequisite name="EvAnnotation"/>
+    <property name="has-popup" type="gboolean" flags="3" default-value="TRUE"/>
+    <property name="label" type="gchararray" flags="3" default-value="NULL"/>
+    <property name="opacity" type="gdouble" flags="3" default-value="1.000000"/>
+    <property name="popup-is-open" type="gboolean" flags="3" default-value="FALSE"/>
+    <property name="rectangle" type="EvRectangle" flags="3"/>
+  </interface>
+  <class name="EvAnnotationText" get-type="ev_annotation_text_get_type" parents="EvAnnotation,GObject">
+    <implements name="EvAnnotationMarkup"/>
+    <property name="icon" type="EvAnnotationTextIcon" flags="3" default-value="EV_ANNOTATION_TEXT_ICON_NOTE"/>
+    <property name="is-open" type="gboolean" flags="3" default-value="FALSE"/>
+  </class>
+  <class name="EvAnnotationAttachment" get-type="ev_annotation_attachment_get_type" parents="EvAnnotation,GObject">
+    <implements name="EvAnnotationMarkup"/>
+    <property name="attachment" type="EvAttachment" flags="7"/>
+  </class>
+  <interface name="EvAsyncRenderer" get-type="ev_async_renderer_get_type">
+    <signal name="render-finished" return="void" when="last">
+      <param type="GdkPixbuf"/>
+    </signal>
+  </interface>
+  <boxed name="EvMappingList" get-type="ev_mapping_list_get_type"/>
+  <interface name="EvDocumentAnnotations" get-type="ev_document_annotations_get_type">
+  </interface>
+  <interface name="EvDocumentAttachments" get-type="ev_document_attachments_get_type">
+  </interface>
+  <interface name="EvDocumentFind" get-type="ev_document_find_get_type">
+  </interface>
+  <class name="EvLinkDest" get-type="ev_link_dest_get_type" parents="GObject">
+    <property name="type" type="EvLinkDestType" flags="11" default-value="EV_LINK_DEST_TYPE_UNKNOWN"/>
+    <property name="page" type="gint" flags="11" default-value="0"/>
+    <property name="left" type="gdouble" flags="11" default-value="0.000000"/>
+    <property name="top" type="gdouble" flags="11" default-value="0.000000"/>
+    <property name="bottom" type="gdouble" flags="11" default-value="0.000000"/>
+    <property name="right" type="gdouble" flags="11" default-value="0.000000"/>
+    <property name="zoom" type="gdouble" flags="11" default-value="0.000000"/>
+    <property name="change" type="guint" flags="11" default-value="0"/>
+    <property name="named" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="page-label" type="gchararray" flags="11" default-value="NULL"/>
+  </class>
+  <class name="EvLinkAction" get-type="ev_link_action_get_type" parents="GObject">
+    <property name="type" type="EvLinkActionType" flags="11" default-value="EV_LINK_ACTION_TYPE_GOTO_DEST"/>
+    <property name="dest" type="gpointer" flags="11"/>
+    <property name="uri" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="filename" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="params" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="name" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="show-list" type="gpointer" flags="11"/>
+    <property name="hide-list" type="gpointer" flags="11"/>
+    <property name="toggle-list" type="gpointer" flags="11"/>
+  </class>
+  <class name="EvLink" get-type="ev_link_get_type" parents="GObject">
+    <property name="title" type="gchararray" flags="11" default-value="NULL"/>
+    <property name="action" type="gpointer" flags="11"/>
+  </class>
+  <interface name="EvDocumentFonts" get-type="ev_document_fonts_get_type">
+  </interface>
+  <class name="EvFormField" get-type="ev_form_field_get_type" parents="GObject" abstract="1">
+  </class>
+  <class name="EvFormFieldText" get-type="ev_form_field_text_get_type" parents="EvFormField,GObject">
+  </class>
+  <class name="EvFormFieldButton" get-type="ev_form_field_button_get_type" parents="EvFormField,GObject">
+  </class>
+  <class name="EvFormFieldChoice" get-type="ev_form_field_choice_get_type" parents="EvFormField,GObject">
+  </class>
+  <class name="EvFormFieldSignature" get-type="ev_form_field_signature_get_type" parents="EvFormField,GObject">
+  </class>
+  <interface name="EvDocumentForms" get-type="ev_document_forms_get_type">
+  </interface>
+  <class name="EvImage" get-type="ev_image_get_type" parents="GObject">
+  </class>
+  <interface name="EvDocumentImages" get-type="ev_document_images_get_type">
+  </interface>
+  <class name="EvLayer" get-type="ev_layer_get_type" parents="GObject">
+  </class>
+  <interface name="EvDocumentLayers" get-type="ev_document_layers_get_type">
+  </interface>
+  <interface name="EvDocumentLinks" get-type="ev_document_links_get_type">
+  </interface>
+  <interface name="EvDocumentPrint" get-type="ev_document_print_get_type">
+  </interface>
+  <interface name="EvDocumentSecurity" get-type="ev_document_security_get_type">
+  </interface>
+  <interface name="EvDocumentThumbnails" get-type="ev_document_thumbnails_get_type">
+  </interface>
+  <class name="EvTransitionEffect" get-type="ev_transition_effect_get_type" parents="GObject">
+    <property name="type" type="EvTransitionEffectType" flags="3" default-value="EV_TRANSITION_EFFECT_REPLACE"/>
+    <property name="alignment" type="EvTransitionEffectAlignment" flags="3" default-value="EV_TRANSITION_ALIGNMENT_HORIZONTAL"/>
+    <property name="direction" type="EvTransitionEffectDirection" flags="3" default-value="EV_TRANSITION_DIRECTION_INWARD"/>
+    <property name="duration" type="gint" flags="3" default-value="0"/>
+    <property name="angle" type="gint" flags="3" default-value="0"/>
+    <property name="scale" type="gdouble" flags="3" default-value="1.000000"/>
+    <property name="rectangular" type="gboolean" flags="3" default-value="FALSE"/>
+  </class>
+  <interface name="EvDocumentTransition" get-type="ev_document_transition_get_type">
+  </interface>
+  <interface name="EvDocumentText" get-type="ev_document_text_get_type">
+  </interface>
+  <interface name="EvFileExporter" get-type="ev_file_exporter_get_type">
+  </interface>
+  <interface name="EvSelection" get-type="ev_selection_get_type">
+  </interface>
+  <enum name="EvAnnotationType" get-type="ev_annotation_type_get_type">
+    <member name="EV_ANNOTATION_TYPE_UNKNOWN" nick="unknown" value="0"/>
+    <member name="EV_ANNOTATION_TYPE_TEXT" nick="text" value="1"/>
+    <member name="EV_ANNOTATION_TYPE_ATTACHMENT" nick="attachment" value="2"/>
+  </enum>  <enum name="EvAnnotationTextIcon" get-type="ev_annotation_text_icon_get_type">
+    <member name="EV_ANNOTATION_TEXT_ICON_NOTE" nick="note" value="0"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_COMMENT" nick="comment" value="1"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_KEY" nick="key" value="2"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_HELP" nick="help" value="3"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_NEW_PARAGRAPH" nick="new-paragraph" value="4"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_PARAGRAPH" nick="paragraph" value="5"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_INSERT" nick="insert" value="6"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_CROSS" nick="cross" value="7"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_CIRCLE" nick="circle" value="8"/>
+    <member name="EV_ANNOTATION_TEXT_ICON_UNKNOWN" nick="unknown" value="9"/>
+  </enum>  <flags name="EvAnnotationsSaveMask" get-type="ev_annotations_save_mask_get_type">
+    <member name="EV_ANNOTATIONS_SAVE_NONE" nick="none" value="0"/>
+    <member name="EV_ANNOTATIONS_SAVE_CONTENTS" nick="contents" value="1"/>
+    <member name="EV_ANNOTATIONS_SAVE_COLOR" nick="color" value="2"/>
+    <member name="EV_ANNOTATIONS_SAVE_LABEL" nick="label" value="4"/>
+    <member name="EV_ANNOTATIONS_SAVE_OPACITY" nick="opacity" value="8"/>
+    <member name="EV_ANNOTATIONS_SAVE_POPUP_RECT" nick="popup-rect" value="16"/>
+    <member name="EV_ANNOTATIONS_SAVE_POPUP_IS_OPEN" nick="popup-is-open" value="32"/>
+    <member name="EV_ANNOTATIONS_SAVE_TEXT_IS_OPEN" nick="text-is-open" value="64"/>
+    <member name="EV_ANNOTATIONS_SAVE_TEXT_ICON" nick="text-icon" value="128"/>
+    <member name="EV_ANNOTATIONS_SAVE_ATTACHMENT" nick="attachment" value="256"/>
+    <member name="EV_ANNOTATIONS_SAVE_ALL" nick="all" value="511"/>
+  </flags>
+  <enum name="EvDocumentLayout" get-type="ev_document_layout_get_type">
+    <member name="EV_DOCUMENT_LAYOUT_SINGLE_PAGE" nick="single-page" value="0"/>
+    <member name="EV_DOCUMENT_LAYOUT_ONE_COLUMN" nick="one-column" value="1"/>
+    <member name="EV_DOCUMENT_LAYOUT_TWO_COLUMN_LEFT" nick="two-column-left" value="2"/>
+    <member name="EV_DOCUMENT_LAYOUT_TWO_COLUMN_RIGHT" nick="two-column-right" value="3"/>
+    <member name="EV_DOCUMENT_LAYOUT_TWO_PAGE_LEFT" nick="two-page-left" value="4"/>
+    <member name="EV_DOCUMENT_LAYOUT_TWO_PAGE_RIGHT" nick="two-page-right" value="5"/>
+  </enum>  <enum name="EvDocumentMode" get-type="ev_document_mode_get_type">
+    <member name="EV_DOCUMENT_MODE_NONE" nick="none" value="0"/>
+    <member name="EV_DOCUMENT_MODE_USE_OC" nick="use-oc" value="1"/>
+    <member name="EV_DOCUMENT_MODE_USE_THUMBS" nick="use-thumbs" value="2"/>
+    <member name="EV_DOCUMENT_MODE_FULL_SCREEN" nick="full-screen" value="3"/>
+    <member name="EV_DOCUMENT_MODE_USE_ATTACHMENTS" nick="use-attachments" value="4"/>
+    <member name="EV_DOCUMENT_MODE_PRESENTATION" nick="presentation" value="3"/>
+  </enum>  <flags name="EvDocumentUIHints" get-type="ev_document_ui_hints_get_type">
+    <member name="EV_DOCUMENT_UI_HINT_HIDE_TOOLBAR" nick="hide-toolbar" value="1"/>
+    <member name="EV_DOCUMENT_UI_HINT_HIDE_MENUBAR" nick="hide-menubar" value="2"/>
+    <member name="EV_DOCUMENT_UI_HINT_HIDE_WINDOWUI" nick="hide-windowui" value="4"/>
+    <member name="EV_DOCUMENT_UI_HINT_FIT_WINDOW" nick="fit-window" value="8"/>
+    <member name="EV_DOCUMENT_UI_HINT_CENTER_WINDOW" nick="center-window" value="16"/>
+    <member name="EV_DOCUMENT_UI_HINT_DISPLAY_DOC_TITLE" nick="display-doc-title" value="32"/>
+    <member name="EV_DOCUMENT_UI_HINT_DIRECTION_RTL" nick="direction-rtl" value="64"/>
+  </flags>
+  <flags name="EvDocumentPermissions" get-type="ev_document_permissions_get_type">
+    <member name="EV_DOCUMENT_PERMISSIONS_OK_TO_PRINT" nick="ok-to-print" value="1"/>
+    <member name="EV_DOCUMENT_PERMISSIONS_OK_TO_MODIFY" nick="ok-to-modify" value="2"/>
+    <member name="EV_DOCUMENT_PERMISSIONS_OK_TO_COPY" nick="ok-to-copy" value="4"/>
+    <member name="EV_DOCUMENT_PERMISSIONS_OK_TO_ADD_NOTES" nick="ok-to-add-notes" value="8"/>
+    <member name="EV_DOCUMENT_PERMISSIONS_FULL" nick="full" value="15"/>
+  </flags>
+  <flags name="EvDocumentInfoFields" get-type="ev_document_info_fields_get_type">
+    <member name="EV_DOCUMENT_INFO_TITLE" nick="title" value="1"/>
+    <member name="EV_DOCUMENT_INFO_FORMAT" nick="format" value="2"/>
+    <member name="EV_DOCUMENT_INFO_AUTHOR" nick="author" value="4"/>
+    <member name="EV_DOCUMENT_INFO_SUBJECT" nick="subject" value="8"/>
+    <member name="EV_DOCUMENT_INFO_KEYWORDS" nick="keywords" value="16"/>
+    <member name="EV_DOCUMENT_INFO_LAYOUT" nick="layout" value="32"/>
+    <member name="EV_DOCUMENT_INFO_CREATOR" nick="creator" value="64"/>
+    <member name="EV_DOCUMENT_INFO_PRODUCER" nick="producer" value="128"/>
+    <member name="EV_DOCUMENT_INFO_CREATION_DATE" nick="creation-date" value="256"/>
+    <member name="EV_DOCUMENT_INFO_MOD_DATE" nick="mod-date" value="512"/>
+    <member name="EV_DOCUMENT_INFO_LINEARIZED" nick="linearized" value="1024"/>
+    <member name="EV_DOCUMENT_INFO_START_MODE" nick="start-mode" value="2048"/>
+    <member name="EV_DOCUMENT_INFO_UI_HINTS" nick="ui-hints" value="4096"/>
+    <member name="EV_DOCUMENT_INFO_PERMISSIONS" nick="permissions" value="8192"/>
+    <member name="EV_DOCUMENT_INFO_N_PAGES" nick="n-pages" value="16384"/>
+    <member name="EV_DOCUMENT_INFO_SECURITY" nick="security" value="32768"/>
+    <member name="EV_DOCUMENT_INFO_PAPER_SIZE" nick="paper-size" value="65536"/>
+    <member name="EV_DOCUMENT_INFO_LICENSE" nick="license" value="131072"/>
+  </flags>
+  <enum name="EvDocumentError" get-type="ev_document_error_get_type">
+    <member name="EV_DOCUMENT_ERROR_INVALID" nick="invalid" value="0"/>
+    <member name="EV_DOCUMENT_ERROR_UNSUPPORTED_CONTENT" nick="unsupported-content" value="1"/>
+    <member name="EV_DOCUMENT_ERROR_ENCRYPTED" nick="encrypted" value="2"/>
+  </enum>  <enum name="EvFileExporterFormat" get-type="ev_file_exporter_format_get_type">
+    <member name="EV_FILE_FORMAT_UNKNOWN" nick="unknown" value="0"/>
+    <member name="EV_FILE_FORMAT_PS" nick="ps" value="1"/>
+    <member name="EV_FILE_FORMAT_PDF" nick="pdf" value="2"/>
+  </enum>  <flags name="EvFileExporterCapabilities" get-type="ev_file_exporter_capabilities_get_type">
+    <member name="EV_FILE_EXPORTER_CAN_PAGE_SET" nick="page-set" value="1"/>
+    <member name="EV_FILE_EXPORTER_CAN_COPIES" nick="copies" value="2"/>
+    <member name="EV_FILE_EXPORTER_CAN_COLLATE" nick="collate" value="4"/>
+    <member name="EV_FILE_EXPORTER_CAN_REVERSE" nick="reverse" value="8"/>
+    <member name="EV_FILE_EXPORTER_CAN_SCALE" nick="scale" value="16"/>
+    <member name="EV_FILE_EXPORTER_CAN_GENERATE_PDF" nick="generate-pdf" value="32"/>
+    <member name="EV_FILE_EXPORTER_CAN_GENERATE_PS" nick="generate-ps" value="64"/>
+    <member name="EV_FILE_EXPORTER_CAN_PREVIEW" nick="preview" value="128"/>
+    <member name="EV_FILE_EXPORTER_CAN_NUMBER_UP" nick="number-up" value="256"/>
+  </flags>
+  <enum name="EvCompressionType" get-type="ev_compression_type_get_type">
+    <member name="EV_COMPRESSION_NONE" nick="none" value="0"/>
+    <member name="EV_COMPRESSION_BZIP2" nick="bzip2" value="1"/>
+    <member name="EV_COMPRESSION_GZIP" nick="gzip" value="2"/>
+    <member name="EV_COMPRESSION_LZMA" nick="lzma" value="3"/>
+  </enum>  <enum name="EvFormFieldTextType" get-type="ev_form_field_text_type_get_type">
+    <member name="EV_FORM_FIELD_TEXT_NORMAL" nick="normal" value="0"/>
+    <member name="EV_FORM_FIELD_TEXT_MULTILINE" nick="multiline" value="1"/>
+    <member name="EV_FORM_FIELD_TEXT_FILE_SELECT" nick="file-select" value="2"/>
+  </enum>  <enum name="EvFormFieldButtonType" get-type="ev_form_field_button_type_get_type">
+    <member name="EV_FORM_FIELD_BUTTON_PUSH" nick="push" value="0"/>
+    <member name="EV_FORM_FIELD_BUTTON_CHECK" nick="check" value="1"/>
+    <member name="EV_FORM_FIELD_BUTTON_RADIO" nick="radio" value="2"/>
+  </enum>  <enum name="EvFormFieldChoiceType" get-type="ev_form_field_choice_type_get_type">
+    <member name="EV_FORM_FIELD_CHOICE_COMBO" nick="combo" value="0"/>
+    <member name="EV_FORM_FIELD_CHOICE_LIST" nick="list" value="1"/>
+  </enum>  <enum name="EvLinkActionType" get-type="ev_link_action_type_get_type">
+    <member name="EV_LINK_ACTION_TYPE_GOTO_DEST" nick="goto-dest" value="0"/>
+    <member name="EV_LINK_ACTION_TYPE_GOTO_REMOTE" nick="goto-remote" value="1"/>
+    <member name="EV_LINK_ACTION_TYPE_EXTERNAL_URI" nick="external-uri" value="2"/>
+    <member name="EV_LINK_ACTION_TYPE_LAUNCH" nick="launch" value="3"/>
+    <member name="EV_LINK_ACTION_TYPE_NAMED" nick="named" value="4"/>
+    <member name="EV_LINK_ACTION_TYPE_LAYERS_STATE" nick="layers-state" value="5"/>
+  </enum>  <enum name="EvLinkDestType" get-type="ev_link_dest_type_get_type">
+    <member name="EV_LINK_DEST_TYPE_PAGE" nick="page" value="0"/>
+    <member name="EV_LINK_DEST_TYPE_XYZ" nick="xyz" value="1"/>
+    <member name="EV_LINK_DEST_TYPE_FIT" nick="fit" value="2"/>
+    <member name="EV_LINK_DEST_TYPE_FITH" nick="fith" value="3"/>
+    <member name="EV_LINK_DEST_TYPE_FITV" nick="fitv" value="4"/>
+    <member name="EV_LINK_DEST_TYPE_FITR" nick="fitr" value="5"/>
+    <member name="EV_LINK_DEST_TYPE_NAMED" nick="named" value="6"/>
+    <member name="EV_LINK_DEST_TYPE_PAGE_LABEL" nick="page-label" value="7"/>
+    <member name="EV_LINK_DEST_TYPE_HLINK" nick="hlink" value="8"/>
+    <member name="EV_LINK_DEST_TYPE_UNKNOWN" nick="unknown" value="9"/>
+  </enum>  <enum name="EvSelectionStyle" get-type="ev_selection_style_get_type">
+    <member name="EV_SELECTION_STYLE_GLYPH" nick="glyph" value="0"/>
+    <member name="EV_SELECTION_STYLE_WORD" nick="word" value="1"/>
+    <member name="EV_SELECTION_STYLE_LINE" nick="line" value="2"/>
+  </enum>  <enum name="EvTransitionEffectType" get-type="ev_transition_effect_type_get_type">
+    <member name="EV_TRANSITION_EFFECT_REPLACE" nick="replace" value="0"/>
+    <member name="EV_TRANSITION_EFFECT_SPLIT" nick="split" value="1"/>
+    <member name="EV_TRANSITION_EFFECT_BLINDS" nick="blinds" value="2"/>
+    <member name="EV_TRANSITION_EFFECT_BOX" nick="box" value="3"/>
+    <member name="EV_TRANSITION_EFFECT_WIPE" nick="wipe" value="4"/>
+    <member name="EV_TRANSITION_EFFECT_DISSOLVE" nick="dissolve" value="5"/>
+    <member name="EV_TRANSITION_EFFECT_GLITTER" nick="glitter" value="6"/>
+    <member name="EV_TRANSITION_EFFECT_FLY" nick="fly" value="7"/>
+    <member name="EV_TRANSITION_EFFECT_PUSH" nick="push" value="8"/>
+    <member name="EV_TRANSITION_EFFECT_COVER" nick="cover" value="9"/>
+    <member name="EV_TRANSITION_EFFECT_UNCOVER" nick="uncover" value="10"/>
+    <member name="EV_TRANSITION_EFFECT_FADE" nick="fade" value="11"/>
+  </enum>  <enum name="EvTransitionEffectAlignment" get-type="ev_transition_effect_alignment_get_type">
+    <member name="EV_TRANSITION_ALIGNMENT_HORIZONTAL" nick="horizontal" value="0"/>
+    <member name="EV_TRANSITION_ALIGNMENT_VERTICAL" nick="vertical" value="1"/>
+  </enum>  <enum name="EvTransitionEffectDirection" get-type="ev_transition_effect_direction_get_type">
+    <member name="EV_TRANSITION_DIRECTION_INWARD" nick="inward" value="0"/>
+    <member name="EV_TRANSITION_DIRECTION_OUTWARD" nick="outward" value="1"/>
+  </enum>  <error-quark function="ev_document_error_quark" domain="ev-document-error-quark"/>
+  <error-quark function="ev_attachment_error_quark" domain="ev-attachment-error-quark"/>
+</dump>

--- a/x11-packages/atril/gir/AtrilView-1.5.0.xml
+++ b/x11-packages/atril/gir/AtrilView-1.5.0.xml
@@ -1,0 +1,161 @@
+<?xml version="1.0"?>
+<dump>
+  <class name="EvDocumentModel" get-type="ev_document_model_get_type" parents="GObject">
+    <property name="document" type="EvDocument" flags="3"/>
+    <property name="page" type="gint" flags="3" default-value="-1"/>
+    <property name="rotation" type="gint" flags="3" default-value="0"/>
+    <property name="inverted-colors" type="gboolean" flags="3" default-value="FALSE"/>
+    <property name="scale" type="gdouble" flags="3" default-value="1.000000"/>
+    <property name="sizing-mode" type="EvSizingMode" flags="3" default-value="EV_SIZING_FIT_WIDTH"/>
+    <property name="continuous" type="gboolean" flags="3" default-value="TRUE"/>
+    <property name="dual-page" type="gboolean" flags="3" default-value="FALSE"/>
+    <property name="dual-odd-left" type="gboolean" flags="3" default-value="FALSE"/>
+    <property name="fullscreen" type="gboolean" flags="3" default-value="FALSE"/>
+    <property name="page-layout" type="EvPageLayout" flags="227" default-value="EV_PAGE_LAYOUT_SINGLE"/>
+    <signal name="page-changed" return="void" when="last">
+      <param type="gint"/>
+      <param type="gint"/>
+    </signal>
+  </class>
+  <class name="EvJob" get-type="ev_job_get_type" parents="GObject" abstract="1">
+    <signal name="cancelled" return="void" when="last">
+    </signal>
+    <signal name="finished" return="void" when="first">
+    </signal>
+  </class>
+  <class name="EvJobLinks" get-type="ev_job_links_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobAttachments" get-type="ev_job_attachments_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobAnnots" get-type="ev_job_annots_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobRender" get-type="ev_job_render_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobPageData" get-type="ev_job_page_data_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobThumbnail" get-type="ev_job_thumbnail_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobFonts" get-type="ev_job_fonts_get_type" parents="EvJob,GObject">
+    <signal name="updated" return="void" when="last">
+      <param type="gdouble"/>
+    </signal>
+  </class>
+  <class name="EvJobLoad" get-type="ev_job_load_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobSave" get-type="ev_job_save_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobFind" get-type="ev_job_find_get_type" parents="EvJob,GObject">
+    <signal name="updated" return="void" when="last">
+      <param type="gint"/>
+    </signal>
+  </class>
+  <class name="EvJobLayers" get-type="ev_job_layers_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobExport" get-type="ev_job_export_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvJobPrint" get-type="ev_job_print_get_type" parents="EvJob,GObject">
+  </class>
+  <class name="EvPrintOperation" get-type="ev_print_operation_get_type" parents="GObject" abstract="1">
+    <property name="document" type="EvDocument" flags="10"/>
+    <signal name="done" return="void" when="last">
+      <param type="GtkPrintOperationResult"/>
+    </signal>
+    <signal name="begin-print" return="void" when="last">
+    </signal>
+    <signal name="status-changed" return="void" when="last">
+    </signal>
+  </class>
+  <class name="EvView" get-type="ev_view_get_type" parents="GtkContainer,GtkWidget,GInitiallyUnowned,GObject">
+    <implements name="AtkImplementorIface"/>
+    <implements name="GtkBuildable"/>
+    <implements name="GtkScrollable"/>
+    <property name="is-loading" type="gboolean" flags="225" default-value="FALSE"/>
+    <signal name="scroll" return="void" when="last" action="1">
+      <param type="GtkScrollType"/>
+      <param type="GtkOrientation"/>
+    </signal>
+    <signal name="handle-link" return="void" when="last" action="1">
+      <param type="GObject"/>
+    </signal>
+    <signal name="external-link" return="void" when="last" action="1">
+      <param type="GObject"/>
+    </signal>
+    <signal name="popup" return="void" when="last" action="1">
+      <param type="gpointer"/>
+    </signal>
+    <signal name="selection-changed" return="void" when="last" action="1">
+    </signal>
+    <signal name="sync-source" return="void" when="last" action="1">
+      <param type="gpointer"/>
+    </signal>
+    <signal name="annot-added" return="void" when="last" action="1">
+      <param type="EvAnnotation"/>
+    </signal>
+    <signal name="annot-removed" return="void" when="last" action="1">
+      <param type="EvAnnotation"/>
+    </signal>
+    <signal name="layers-changed" return="void" when="last" action="1">
+    </signal>
+    <signal name="move-cursor" return="gboolean" when="last" action="1">
+      <param type="GtkMovementStep"/>
+      <param type="gint"/>
+      <param type="gboolean"/>
+    </signal>
+    <signal name="cursor-moved" return="void" when="last">
+      <param type="gint"/>
+      <param type="gint"/>
+    </signal>
+  </class>
+  <class name="EvWebView" get-type="ev_web_view_get_type" parents="WebKitWebView,WebKitWebViewBase,GtkContainer,GtkWidget,GInitiallyUnowned,GObject">
+    <implements name="AtkImplementorIface"/>
+    <implements name="GtkBuildable"/>
+  </class>
+  <class name="EvViewPresentation" get-type="ev_view_presentation_get_type" parents="GtkWidget,GInitiallyUnowned,GObject">
+    <implements name="AtkImplementorIface"/>
+    <implements name="GtkBuildable"/>
+    <property name="document" type="EvDocument" flags="10"/>
+    <property name="current-page" type="guint" flags="6" default-value="0"/>
+    <property name="rotation" type="guint" flags="7" default-value="0"/>
+    <property name="inverted-colors" type="gboolean" flags="10" default-value="FALSE"/>
+    <signal name="finished" return="void" when="last" action="1">
+    </signal>
+    <signal name="external-link" return="void" when="last" action="1">
+      <param type="GObject"/>
+    </signal>
+    <signal name="change-page" return="void" when="last" action="1">
+      <param type="GtkScrollType"/>
+    </signal>
+  </class>
+  <enum name="EvSizingMode" get-type="ev_sizing_mode_get_type">
+    <member name="EV_SIZING_FIT_PAGE" nick="fit-page" value="0"/>
+    <member name="EV_SIZING_BEST_FIT" nick="best-fit" value="0"/>
+    <member name="EV_SIZING_FIT_WIDTH" nick="fit-width" value="1"/>
+    <member name="EV_SIZING_FREE" nick="free" value="2"/>
+    <member name="EV_SIZING_AUTOMATIC" nick="automatic" value="3"/>
+  </enum>  <enum name="EvPageLayout" get-type="ev_page_layout_get_type">
+    <member name="EV_PAGE_LAYOUT_SINGLE" nick="single" value="0"/>
+    <member name="EV_PAGE_LAYOUT_DUAL" nick="dual" value="1"/>
+    <member name="EV_PAGE_LAYOUT_AUTOMATIC" nick="automatic" value="2"/>
+  </enum>  <enum name="EvJobPriority" get-type="ev_job_priority_get_type">
+    <member name="EV_JOB_PRIORITY_URGENT" nick="priority-urgent" value="0"/>
+    <member name="EV_JOB_PRIORITY_HIGH" nick="priority-high" value="1"/>
+    <member name="EV_JOB_PRIORITY_LOW" nick="priority-low" value="2"/>
+    <member name="EV_JOB_PRIORITY_NONE" nick="priority-none" value="3"/>
+    <member name="EV_JOB_N_PRIORITIES" nick="n-priorities" value="4"/>
+  </enum>  <enum name="EvJobRunMode" get-type="ev_job_run_mode_get_type">
+    <member name="EV_JOB_RUN_THREAD" nick="thread" value="0"/>
+    <member name="EV_JOB_RUN_MAIN_LOOP" nick="main-loop" value="1"/>
+  </enum>  <flags name="EvJobPageDataFlags" get-type="ev_job_page_data_flags_get_type">
+    <member name="EV_PAGE_DATA_INCLUDE_NONE" nick="none" value="0"/>
+    <member name="EV_PAGE_DATA_INCLUDE_LINKS" nick="links" value="1"/>
+    <member name="EV_PAGE_DATA_INCLUDE_TEXT" nick="text" value="2"/>
+    <member name="EV_PAGE_DATA_INCLUDE_TEXT_MAPPING" nick="text-mapping" value="4"/>
+    <member name="EV_PAGE_DATA_INCLUDE_TEXT_LAYOUT" nick="text-layout" value="8"/>
+    <member name="EV_PAGE_DATA_INCLUDE_TEXT_ATTRS" nick="text-attrs" value="16"/>
+    <member name="EV_PAGE_DATA_INCLUDE_TEXT_LOG_ATTRS" nick="text-log-attrs" value="32"/>
+    <member name="EV_PAGE_DATA_INCLUDE_IMAGES" nick="images" value="64"/>
+    <member name="EV_PAGE_DATA_INCLUDE_FORMS" nick="forms" value="128"/>
+    <member name="EV_PAGE_DATA_INCLUDE_ANNOTS" nick="annots" value="256"/>
+    <member name="EV_PAGE_DATA_INCLUDE_ALL" nick="all" value="511"/>
+  </flags>
+</dump>

